modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+----
+
+Because the same constraint already exists, nothing will happen:
+
+.Result
+[source, queryresult]
+----
+(no changes, no records)
+----
+
+.Notification
+[source]
+----
+`CREATE CONSTRAINT sequels IF NOT EXISTS FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order) IS UNIQUE` has no effect.
+`CONSTRAINT sequels FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order, e.seriesTitle) IS UNIQUE` already exists.
+----
+
+======
+
+
+[[constraint-creation-failures-and-data-violations]]
+=== Constraint creation failures and data violation scenarios
+
+* xref:constraints/managing-constraints.adoc#create-an-already-existing-constraint[]
+* xref:constraints/managing-constraints.adoc#create-data-that-violates-a-constraint[]
+* xref:constraints/managing-constraints.adoc#fail-to-create-constraint-due-to-existing-data[]
+* xref:constraints/managing-constraints.adoc#removing-an-existing-constrained-property-will-fail[]
+* xref:constraints/managing-constraints.adoc#create-key-and-uniqueness-constraint-on-same-schema-as-existing-index[]
+
+
+[[create-an-already-existing-constraint]]
+==== Creating an already existing constraint will fail
+
+Creating a constraint on a node label or relationship property that is already constrained will fail.
+This applies to all constraint types.
+
+.Create an already existing node property uniqueness constraint
+======
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE CONSTRAINT preExisting_book_published FOR (book:Book) REQUIRE book.published IS UNIQUE
+----
+////
+
+.Create a constraint requiring `Book` nodes to have unique `published` properties, when that constraint already exists
+[source, cypher, role=test-fail]
+----
+CREATE CONSTRAINT book_published FOR (book:Book) REQUIRE book.published IS UNIQUE
+----
+
+In this case, the constraint cannot be created because it already exists.
+
+.Error message
+[source, error]
+----
+Constraint already exists:
+Constraint( id=4, name='preExisting_book_published', type='UNIQUENESS', schema=(:Book {published}), ownedIndex=3 )
+----
+
+[NOTE]
+====
+The constraint type will be updated to say `NODE PROPERTY UNIQUENESS` in Neo4j 6.0.
+====
+
+======
+
+.Create an already existing relationship property type constraint
+======
+
+
+.Create a constraint requiring `order` properties on `PART_OF` relationships to be of type `INTEGER`, when that constraint already exists
+[source, cypher, role=test-fail]
+----
+CREATE CONSTRAINT belongs_to
+FOR ()-[part:PART_OF]-() REQUIRE part.order :: INTEGER
+----
+
+In this case, the constraint cannot be created because it already exists (but with a different name).
+
+.Error message
+[source, error]
+----
+Constraint already exists: Constraint( id=24, name='part_of', type='RELATIONSHIP PROPERTY TYPE', schema=()-[:PART_OF {order}]-(), propertyType=INTEGER )
+----
+
+======
+
+
+[[create-data-that-violates-a-constraint]]
+==== Creating data that violates existing constraints will fail
+
+This applies to all constraint types.
```

Comment:
I think it might be nice to list how it applies to the different constraint types as they will all fail on different things? so it feels weird to me to only say "it applies to all" as if all examples applies to all constraint types...

Reply:
That would especially be nice since we only show one of the ways the key constraint can force your query to fail for example

Reply:
I have in one of my later comments a suggestion for making it a table to show what things affect each constraint type, that would probably be nice here as well